File: sprint04/JMS/PGS_42833.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "", temp = "";
    int i=0,j=k;
    
    while(i < k && j > 0){
        int max = -1, maxindex= -1;
        
        for(int q=i;q<j+i+1;q++){
            if(max < number[q]){
                max = number[q];
                maxindex = q;
            }
        }
        
        temp = number.substr(0, i);
        temp += number.substr(maxindex);
        j = j - maxindex + i;
        i++;
        number = temp;
    }
    
    if(j>0)
        number = number.substr(0, number.length() - j);
    
    return number;
}

File: sprint04/KDH/01/PGS_42833.py
def solution(number, k):
    answer = []
    for num in number:
        if not answer:
            answer.append(num)
            continue

        if k > 0:
            while answer:
                if answer[-1] < num:
                    answer.pop()
                    k -= 1
                    if k == 0:
                        break
                else:
                    break

        answer.append(num)

    answer = ''.join(answer[:len(number)-k])

    return answer
